stratified vs. random cross-val.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jul 14 18:29:08 2021
+
+@author: matthewayala
+"""
+
+import numpy as np
+from sklearn.linear_model import LogisticRegression
+import matplotlib.pyplot as plt
+from sklearn.model_selection import StratifiedKFold
+from sklearn.model_selection import KFold
+
+# Simulate logistic data with/without noise
+
+def logistic(x, b, noise=None):
+    L = x.T.dot(b)
+    if noise is not None:
+        L = L+noise
+    return 1/(1+np.exp(-L))
+
+# Function that sumulates logistic data, trains and validates model on the data.
+
+def get_noise_accuracy(noise_iterable,x_values,cut_off):
+    """Returns the noise iterable, the list of accuracy scores, and the list of standard deviations for error bars."""
+    bias = np.ones(len(x_values))
+    X = np.vstack([x_values,bias]) # Add Intercept
+    B =  [1., 1.] # Sigmoid params for X   
+    
+    mean_acc_scores = []
+    standard_deviations = []
+    noise_list = noise_iterable # Different noise values to be used in gaussian distribution sampling.
+    for i in noise_list:
+        # Creating simulated logistic data set with several noise values
+        pnoisy = logistic(X, B, noise=np.random.normal(loc=0., scale= i, size=len(x_values)))
+        
+        # Reshaping data for model    
+        x_data = x_values.reshape(len(x_values),1)
+        pnoisy_data = pnoisy.reshape(len(x_values),1)
+        transformed_p = [] # Empty list to hold new transformed data (1's and 0's)
+        for a in pnoisy_data:    
+            if a >= cut_off: # Turning values > 0.5 in 1's and < 0.5 into 0's
+                a = 1
+            else:
+                a = 0
+            transformed_p.append(a) # Adding transformed values to the empty list.
+        
+        # Turning labels into an array for stratifiedkfold
+        transformed_p = np.array(transformed_p)
+        # Creating Logistic regression instance
+        LogReg = LogisticRegression(max_iter=1000)
+        
+        # StratifiedKFold Splits
+        skf = StratifiedKFold(n_splits = 5)
```

Comment:
You should specify a random seed here, so that the splits are reproducible:
```suggestion
        skf = StratifiedKFold(n_splits = 5, random_state=SEED)
```
Then you can add `SEED` as a parameter at the top of the file.

Reply:
When I try and add this in I get this error - "Setting a random_state has no effect since shuffle is False. You should leave random_state to its default (None), or set shuffle=True."

I'm not quite sure where the shuffle parameter is? If it is a part of StratifiedKFold or not.

Reply:
Ahh, I see. `shuffle` is a parameter to StratifiedKFold. I'd suggest setting `shuffle=True` and including the random seed.

That parameter tells StratifiedKFold to shuffle the data before splitting it into train and test sets. For your simulated data it doesn't matter that much because you're generating it in (essentially) a random order, but for real datasets the order might be meaningful: say, for example, you collected some cancer samples over time, and their order in the file is from earliest to most recent. You wouldn't want the model to just learn whether samples are old or new, since that pattern isn't likely to generalize well to other datasets. Randomizing (shuffling) the order would get rid of that signal and allow your model to learn more general patterns in the data.

So, in short - I'd generally recommend `shuffle=True`, both here and in the future.

Reply:
Awesome, fixed it up!